slither/detectors/functions/cyclomatic_complexity.py

```diff
@@ -0,0 +1,41 @@
+from slither.detectors.abstract_detector import AbstractDetector, DetectorClassification
+from slither.utils.code_complexity import compute_cyclomatic_complexity
+
+
+class CyclomaticComplexity(AbstractDetector):
+    """
+    Detects functions with high (> 11) cyclomatic complexity.
+    """
+
+    ARGUMENT = "cyclomatic-complexity"
+    HELP = "Detects functions with high (> 11) cyclomatic complexity"
+    IMPACT = DetectorClassification.INFORMATIONAL
+    CONFIDENCE = DetectorClassification.HIGH
+
+    WIKI = 'https://github.com/crytic/slither/wiki/Detector-Documentation#cyclomatic-complexity"'
+
+    WIKI_TITLE = "Cyclomatic complexity detector"
+    WIKI_DESCRIPTION = "Detects functions with high (> 11) cyclomatic complexity."
+    WIKI_EXPLOIT_SCENARIO = ""
+    WIKI_RECOMMENDATION = (
+        "Reduce cyclomatic complexity by splitting the function into several smaller subroutines."
+    )
+
+    def _detect(self):
+        results = []
+
+        for f in self.compilation_unit.functions:
+            cc = compute_cyclomatic_complexity(f)
+            if cc > 11:
+                info = (
+                    "Function "
```

Comment:
We should use `[ f, f" has a high cyclomatic complexity ({cc})"]` here.

`generate_results` has inbuilt that will transform slither object (i.e. `f` here) in a human readable format (ex: adding the source mapping directly)